### ISSUE TITLE
Indexing error

### ISSUE THREAD

**Issue by kcastro18** [OrchardCMS/Orchard#7987]
Why am I getting this error when I try to rebuild my site index?

```
ltOrchardEventBus.TryInvokeMethod(IEventHandler eventHandler, Type interfaceType, String messageName, String interfaceName, String methodName, IDictionary`2 arguments, IEnumerable& returnValue)
   at Orchard.Events.DefaultOrchardEventBus.TryNotifyHandler(IEventHandler eventHandler, String messageName, String interfaceName, String methodName, IDictionary`2 eventData, IEnumerable& returnValue)
2018-03-02 16:35:51,358 [30] Orchard.Exceptions.DefaultExceptionPolicy - (null) - An unexpected exception was caught [http://jud-tweb2/orcharddev/Admin/Indexing/Rebuild/SiteIndex]
System.NullReferenceException: Object reference not set to an instance of an object.
   at Orchard.ContentManagement.DefaultContentManager.Get(Int32 id, VersionOptions options, QueryHints hints)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.<DistinctIterator>d__64`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Orchard.Indexing.Services.IndexingTaskExecutor.BatchIndex(String indexName, String settingsFilename, IndexSettings indexSettings)
   at Orchard.Indexing.Services.IndexingTaskExecutor.UpdateIndexBatch(String indexName)
   at Orchard.Indexing.Services.UpdateIndexScheduler.UpdateIndex(String indexName)
   at Orchard.Events.DelegateHelper.<>c__DisplayClass14_0`2.<BuildAction>b__0(Object target, Object[] p)
   at Orchard.Events.DefaultOrchardEventBus.TryInvokeMethod(IEventHandler eventHandler, Type interfaceType, String messageName, String interfaceName, String methodName, IDictionary`2 arguments, IEnumerable& returnValue)
   at Orchard.Events.DefaultOrchardEventBus.TryNotifyHandler(IEventHandler eventHandler, String messageName, String interfaceName, String methodName, IDictionary`2 eventData, IEnumerable& returnValue)
2018-03-02 16:35:51,952 [9] Orchard.Exceptions.DefaultExceptionPolicy - (null) - An unexpected exception was caught [http://jud-tweb2/orcharddev/Admin/Indexing/Rebuild/SiteIndex]
System.NullReferenceException: Object reference not set to an instance of an object.
   at Orchard.ContentManagement.DefaultContentManager.Get(Int32 id, VersionOptions options, QueryHints hints)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.<DistinctIterator>d__64`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Orchard.Indexing.Services.IndexingTaskExecutor.BatchIndex(String indexName, String settingsFilename, IndexSettings indexSettings)
   at Orchard.Indexing.Services.IndexingTaskExecutor.UpdateIndexBatch(String indexName)
   at Orchard.Indexing.Services.UpdateIndexScheduler.UpdateIndex(String indexName)
   at Orchard.Events.DelegateHelper.<>c__DisplayClass14_0`2.<BuildAction>b__0(Object target, Object[] p)
   at Orchard.Events.DefaultOrchardEventBus.TryInvokeMethod(IEventHandler eventHandler, Type interfaceType, String 
```

**Comment by sebastienros** [comment]
Which version?

**Comment by kcastro18** [comment]
It's 1.10.2

**Comment by sebastienros** [comment]
Fixed in 1.10.x
https://github.com/OrchardCMS/Orchard/blob/1.10.x/src/Orchard.Web/Modules/Orchard.Indexing/Services/IndexingTaskExecutor.cs#L163-L166

You can either use this branch or apply the fix locally
